component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/CasqueAuthenticator.java

```diff
@@ -0,0 +1,261 @@
+            RadiusResponse radiusResponse = Radius.sendRequest("CASQUE SNR", tokenIdPlusName, null);
+            int radiusResponseType = radiusResponse.getType();
+
+            if (radiusResponseType == RadiusResponse.ACCESS_CHALLENGE) { // Got a challenge
+                context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, radiusResponse.getState());
+                String challenge = radiusResponse.getChallenge();
+                String contextIdentifier = context.getContextIdentifier();
+                authPages.challengePage(response, contextIdentifier, challenge);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            }
+
+            clearProperties(context);
+
+            if (radiusResponseType == RadiusResponse.ACCESS_REJECT) {
+                throw new InvalidCredentialsException(
+                        " User authentication failed due to invalid credentials ", User.getUserFromUserName(userName));
+            }
+
+            throw new InvalidCredentialsException(" User authentication failed due to " + radiusResponse.getError(),
+                    User.getUserFromUserName(userName));
+        } catch (CasqueException ce) {
+            throw new AuthenticationFailedException("Unable to start the authentication process", ce);
+        }
+    }
+
+    /**
+     * clean up the properties we set: RADIUS_STATE and USER_NAME
+     */
+    private void clearProperties(AuthenticationContext context) {
+
+        context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, null);
+        context.setProperty(CasqueAuthenticatorConstants.USER_NAME, null);
+    }
+
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context)
+            throws AuthenticationFailedException, LogoutFailedException {
+
+        if (context.isLogoutRequest()) {
+            try {
+                processLogoutResponse(request, response, context);
+            } catch (UnsupportedOperationException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Ignoring UnsupportedOperationException", e);
+                }
+            }
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        }
+        byte[] radiusState = (byte[]) context.getProperty(CasqueAuthenticatorConstants.RADIUS_STATE);
+
+        if (radiusState == null) { // Initial request, get a challenge
+            AuthenticatorFlowStatus status = null;
+            try {
+                status = start(request, response, context);
+            } catch (AuthenticationFailedException e) {
+                throw new AuthenticationFailedException("Authentication failed Accord while sending request ", e);
+            }
+            context.setCurrentAuthenticator(getName());
+            return status;
+        }
+        // radiusState is not null so handle the response to the challenge
```

Comment:
Use the proper commenting format. Check everyplace and fix it.

Reply:
fixed with - https://github.com/wso2-extensions/identity-outbound-auth-casque-snr/pull/8